src/java.base/share/classes/java/lang/classfile/package-info.java

```diff
@@ -356,10 +379,23 @@
  * attributes that are not transformed can be processed by bulk-copying their
  * bytes, rather than parsing them and regenerating their contents.)  If
  * constant pool sharing is not desired it can be suppressed
- * with the {@link jdk.internal.classfile.Classfile.ConstantPoolSharingOption} option.
+ * with the {@link java.lang.classfile.ClassFile.ConstantPoolSharingOption} option.
  * Such suppression may be beneficial when transformation removes many elements,
  * resulting in many unreferenced constant pool entries.
  *
+ * <h3>Transformation handling of unknown classfile elements from a future</h3>
+ * To achieve stability of transformations in future JDK releases it is desired
+ * to set appropriate response to potential new classfile elements appearing.
+ * Transformation designed to handle all situation should be implemented strict way,
```

Comment:
"handle all situations" - I couldn't make much sense of it. I only got it when reading the text that follows, which says "transformation selectively interested in ...". This para should be rephrased for better clarity. E.g.
"Classfile transforms that are interested in consuming _all_ classfile elements should be implemented in a strict fashion, so that new exceptions will be issued, should a new and unknown classfile element appear ... Conversely, classfile transforms that are only interested in consuming a portion of classfile elements do not need to concern with new and unknown classfile elements ... because ... "

Reply:
Thanks for the review, I've reformulated the paragraph as suggested.